### PR TITLE
Feature/xenforo

### PR DESCRIPTION
**Pull request by rkuo-danswer**
## Description
Fixes DAN-669.

## How Has This Been Tested?
[Describe the tests you ran to verify your changes]

## Accepted Risk
[Any know risks or failure modes to point out to reviewers]

## Related Issue(s)
[If applicable, link to the issue(s) this PR addresses]

## Checklist:
- [ ] All of the automated tests pass
- [ ] All PR comments are addressed and marked resolved
- [ ] If there are migrations, they have been rebased to latest main
- [ ] If there are new dependencies, they are added to the requirements
- [ ] If there are new environment variables, they are added to all of the deployment methods
- [ ] If there are new APIs that don't require auth, they are added to PUBLIC_ENDPOINT_SPECS
- [ ] Docker images build and basic functionalities work
- [ ] Author has done a final read through of the PR right before merge

### CODE REVIEW

**Comment by rkuo-danswer** [comment]
Cleaning up https://github.com/danswer-ai/danswer/pull/1038

File: web/src/lib/connectors/connectors.ts [hunk 1]
```diff
@@ -2,6 +2,15 @@ import * as Yup from "yup";
 import { IsPublicGroupSelectorFormType } from "@/components/IsPublicGroupSelector";
 import { ConfigurableSources, ValidInputTypes, ValidSources } from "../types";
 
+export function isLoadState(connector_name: string): boolean {
+  const connectors = ["web", "xenforo"];
```
Comment (pablonyx) [inline web/src/lib/connectors/connectors.ts]: super nit: would prefer a more descriptive variable name

File: backend/danswer/connectors/xenforo/connector.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+            soup = self.requestsite(thread_url)
+            if soup is None:
+                logger.error(f"Failed to load page: {self.base_url}")
+                continue
+            pages = get_pages(soup, thread_url)
+            # Getting all pages for all threads
+            for page_count, page in enumerate(pages, start=1):
+                logger.info(f"Visiting {page}")
+                logger.info(
+                    f"\x1b[KProgress: Page {page_count}/{len(pages)} - Thread {thread_count}/{len(all_threads)}\r"
```
Comment (pablonyx) [inline backend/danswer/connectors/xenforo/connector.py]: Never seen this escape sequence

File: backend/danswer/connectors/xenforo/connector.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+                semantic_identifier=title,
+                primary_owners=[BasicExpertInfo(display_name=author)],
+                metadata={
+                    "type": "post",
+                    "author": author,
+                    "time": formatted_time,
+                },
+                doc_updated_at=post_date,
+            )
+            # logger.info(f"xenforo doc_id: doc_id={document.id} text={post_text}")
```
Comment (pablonyx) [inline backend/danswer/connectors/xenforo/connector.py]: nit: small comment left in

**Review by pablodanswer — APPROVED** [review]
Approved with a small comment

File: backend/danswer/connectors/xenforo/connector.py (hunk 2 above)
Comment (rkuo-danswer) [reply backend/danswer/connectors/xenforo/connector.py]: appears to be an escape sequence to update the same log line on the console. I'll just remove it.

**Pull request merged**